src/composable/scraper.ts

```diff
@@ -147,3 +147,69 @@ export async function getSocialImageUrl(
     return null;
   }
 }
+
+/**
+ * Get favicon URL from HTML link tags or fallback to /favicon.ico.
+ *
+ * This function scrapes the given URL as HTML and tries to extract favicon
+ * URLs from link tags. It supports:
+ * - <link rel="icon" href="...">
+ * - <link rel="shortcut icon" href="...">
+ * - <link rel="apple-touch-icon" href="...">
+ *
+ * When more than one matching tags exists, this will return the last parsed
+ * tag with preference to "icon" rel values.
```

Comment:
Add a note we ignore media/type/sizes attributes from spec and always use the last matching tag here.